src/PlusDataCollection/Atracsys/AtracsysTracker.cxx

```diff
@@ -0,0 +1,799 @@
+  {
+    return ERROR_ENABLE_USED_LED;
+  }
+  return SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+AtracsysTracker::ATRACSYS_RESULT AtracsysTracker::SetMaxMissingFiducials(int maxMissingFids)
+{
+  ftkSetInt32(this->Internal->FtkLib, this->Internal->TrackerSN, OPTION_MAX_MISSING_POINTS, maxMissingFids);
+  return SUCCESS;
+}
+
+// ------------------------------------------
+// spryTrack only options
+// ------------------------------------------
+
+AtracsysTracker::ATRACSYS_RESULT AtracsysTracker::EnableOnboardProcessing(bool enabled)
+{
+  return this->SetSpryTrackOnlyOption(OPTION_ONBOARD_PROCESSING, enabled, ERROR_ENABLE_ONBOARD_PROCESSING);
+}
+
+//----------------------------------------------------------------------------
+AtracsysTracker::ATRACSYS_RESULT AtracsysTracker::EnableImageStreaming(bool enabled)
+{
+  return this->SetSpryTrackOnlyOption(OPTION_IMAGE_STREAMING, enabled, ERROR_ENABLE_IMAGE_STREAMING);
+}
+
+//----------------------------------------------------------------------------
+AtracsysTracker::ATRACSYS_RESULT AtracsysTracker::EnableWirelessMarkerPairing(bool enabled)
+{
+  return this->SetSpryTrackOnlyOption(OPTION_WIRELESS_MARKER_PAIRING, enabled, ERROR_ENABLE_WIRELESS_MARKER_PAIRING);
+}
+
+//----------------------------------------------------------------------------
+AtracsysTracker::ATRACSYS_RESULT AtracsysTracker::EnableWirelessMarkerStatusStreaming(bool enabled)
+{
+  return this->SetSpryTrackOnlyOption(OPTION_WIRELESS_MARKER_PAIRING, enabled, ERROR_ENABLE_WIRELESS_MARKER_STATUS_STREAMING);
+}
+
+//----------------------------------------------------------------------------
+AtracsysTracker::ATRACSYS_RESULT AtracsysTracker::EnableWirelessMarkerBatteryStreaming(bool enabled)
+{
+  return this->SetSpryTrackOnlyOption(OPTION_WIRELESS_MARKER_BATTERY_STREAMING, enabled, ERROR_ENABLE_WIRELESS_MARKER_BATTERY_STREAMING);
+}
+
+// ------------------------------------------
+// fusionTrack only options
+// ------------------------------------------
+
+AtracsysTracker::ATRACSYS_RESULT AtracsysTracker::GetDroppedFrameCount(int& droppedFrameCount)
+{
+  return SUCCESS;
+}
+
+AtracsysTracker::ATRACSYS_RESULT AtracsysTracker::ResetLostFrameCount()
+{
+  return SUCCESS;
+}
+
+// ------------------------------------------
+// Marker methods
+// ------------------------------------------
+
+AtracsysTracker::Marker::Marker(int geometryId, vtkSmartPointer<vtkMatrix4x4> toolToTracker, int gpm, float freMm)
+{
+  this->GeometryId = geometryId;
+  this->ToolToTracker = toolToTracker;
+  this->GeometryPresenceMask = gpm;
+  this->FreMm = freMm;
+}
+
+int AtracsysTracker::Marker::GetGeometryID()
+{
+  return this->GeometryId;
+}
+
+int AtracsysTracker::Marker::GetGeometryPrecsenceMask()
+{
+  return this->GeometryPresenceMask;
+}
+
+vtkSmartPointer<vtkMatrix4x4> AtracsysTracker::Marker::GetTransformToTracker()
```

Comment:
The matrix cannot be deleted before returning, so use vtkMatrix4x4* instead of vtkSmartPointer.